### PR TITLE
domiftune: add 4 cases

### PR DESCRIPTION
**Pull request by yanqzhan**
Add 4 cases in domiftune to complete automation of RHEL7-45987.

Signed-off-by: Yanqiu Zhang <yanqzhan@redhat.com>

### CODE REVIEW

**Comment by yanqzhan** [comment]
Local test passed:
# avocado run --vt-type libvirt --vt-machine-type q35 virsh.domiftune
JOB ID     : cb04470aea6e6aa384067cd8bcc677d268fed30b
JOB LOG    : /root/avocado/job-results/job-2021-07-29T03.56-cb04470/job.log
...
 (13/53) type_specific.io-github-autotest-libvirt.virsh.domiftune.positive_testing.set_domif_parameter.running_guest.change_inbound.options.none.with_floor: PASS (43.91 s)
...
 (26/53) type_specific.io-github-autotest-libvirt.virsh.domiftune.positive_testing.set_domif_parameter.shutoff_guest: PASS (40.33 s)
...
 (28/53) type_specific.io-github-autotest-libvirt.virsh.domiftune.negative_testing.get_domif_parameter.running_guest.options.exclusive: PASS (25.15 s)
...
 (39/53) type_specific.io-github-autotest-libvirt.virsh.domiftune.negative_testing.set_domif_parameter.running_guest.change_inbound.options.none.floor_for_nobandwidth: PASS (183.29 s)
...
RESULTS    : PASS 47 | ERROR 0 | FAIL 6 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1796.75 s

And from the log, steps and results of each case are as expected.

**Comment by yanqzhan** [comment]
The 6 are failed by Bug '1910745'.

**Comment by yanqzhan** [comment]
@yalzhang Please help review if you have time. Thanks.

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py [hunk 1]
```diff
@@ -338,19 +362,55 @@ def run(test, params, env):
     start_vm = params.get("start_vm", "yes")
     change_parameters = params.get("change_parameters", "no")
     interface_ref = params.get("interface_ref", "name")
-
+    netfloor = params.get("netfloor")
```
Comment (yalzhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py:365]: It seems that L365 duplicate with L42.

**Comment by yalzhang** [comment]
suggest to add the manual case id in comment for reference.

**Comment by yanqzhan** [comment]
> comment

Included in commit msg already.

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py (hunk 1 above)
Comment (yanqzhan) [reply libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py:365]: Thanks. But they're in different functions which are at same level. Error occurs if remove it:
 ERROR: name 'netfloor' is not defined
Ps: you can also see many other params duplicated in them, e.g. vm_name.

**Comment by yanqzhan** [comment]
@yalzhang Replied. Pls review again. Thank you!

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py [hunk 2]
```diff
@@ -74,12 +84,15 @@ def check_domiftune(params, test_clear):
...
         burst_outbound_from_cmd_output = int(dicts['outbound.burst'])
+        floor_inbound_from_cmd_output = int(dicts['inbound.floor'])
 
         logging.debug("inbound and outbound from cmd output:")
         logging.debug("inbound: %s,%s,%s; outbound: %s,%s,%s"
                       % (average_inbound_from_cmd_output, peak_inbound_from_cmd_output,
                          burst_inbound_from_cmd_output, average_outbound_from_cmd_output,
                          peak_outbound_from_cmd_output, burst_outbound_from_cmd_output))
+        if netfloor:
+            logging.debug("inbound floor: %s" % floor_inbound_from_cmd_output)
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py:95]: The correct usage for logging.debug() should be 
`  logging.debug("inbound floor: %s", floor_inbound_from_cmd_output)`

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py [hunk 3]
```diff
@@ -74,12 +84,15 @@ def check_domiftune(params, test_clear):
         peak_outbound_from_cmd_output = int(dicts['outbound.peak'])
         burst_inbound_from_cmd_output = int(dicts['inbound.burst'])
         burst_outbound_from_cmd_output = int(dicts['outbound.burst'])
+        floor_inbound_from_cmd_output = int(dicts['inbound.floor'])
 
         logging.debug("inbound and outbound from cmd output:")
         logging.debug("inbound: %s,%s,%s; outbound: %s,%s,%s"
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py:90]: Could you help also fix this line?
The correct usage for logging.debug() should be 
`  logging.debug("inbound floor: %s", floor_inbound_from_cmd_output)`

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py [hunk 4]
```diff
@@ -103,6 +116,9 @@ def check_domiftune(params, test_clear):
             logging.debug("inbound from xml:")
             logging.debug("%s, %s, %s" % (average_inbound_from_xml, peak_inbound_from_xml,
                                           burst_inbound_from_xml))
+            if netfloor:
+                floor_inbound_from_xml = int(domiftune_params.get("inbound").get("floor"))
+                logging.debug("inbound floor from xml: %s" % floor_inbound_from_xml)
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py:121]: The correct usage for logging.debug() should be
logging.debug("inbound floor: %s", floor_inbound_from_cmd_output)

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py [hunk 5]
```diff
@@ -338,19 +362,55 @@ def run(test, params, env):
...
-
+    netfloor = params.get("netfloor")
+    pre_vmstate = params.get("pre_vmstate")
     interface = []
+    change_net = False
+
+    def set_netbw(netfloor, if_net, netxml):
+        """
+        Set network bandwidth as required.
+        :param netfloor: action on network bandwidth, 'need' or 'delete'
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py:373]: Add a blank line above this.

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py [hunk 6]
```diff
@@ -338,19 +362,55 @@ def run(test, params, env):
...
+        elif netfloor == 'need':
+            netxml.bandwidth_inbound = {'average': 100, 'peak': 200, 'burst': 128}
+            netxml.bandwidth_outbound = {'average': 50, 'peak': 100, 'burst': 128}
+        netxml.sync()
+        virsh.net_dumpxml(if_net, debug=True)
+
+    def get_iface(vm_name, if_mac):
+        """
+        Get the first interface dev from guest xml
+        :param vm_name: guest name
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py:388]: Add a blank line above this.

**Review by yalzhang — COMMENTED** [review]
have ran the cases locally and pass, the checkpoints are covered.

**Review by yalzhang — APPROVED** [review]

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py (hunk 2 above)
Comment (yanqzhan) [reply libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py]: Thanks. Done

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py (hunk 3 above)
Comment (yanqzhan) [reply libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py]: Done.

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py (hunk 4 above)
Comment (yanqzhan) [reply libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py]: Done.

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py (hunk 5 above)
Comment (yanqzhan) [reply libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py:374]: Done.

**Comment by yanqzhan** [comment]
@dzhengfy Thanks for your comments. Codes updated. Do you mind review it again? Thanks.

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**